javalib/src/main/scala/java/util/Formatter.scala

```diff
@@ -39,7 +40,13 @@ final class Formatter(private[this] var dest: Appendable)
   private[this] var closed: Boolean = false
   private[this] var lastIOException: IOException = null
 
-  def this() = this(null: Appendable)
+  def this() = this(null: Appendable, Formatter.RootLocaleInfo)
```

Comment:
Shouldn't this (maybe just "technically") be the English locale? JavaDoc says if unspecified, its the default locale, and IIUC, Scala.js specifies this as "en-US". (the relevant test seems to indeed exclude the test for the JVM, even though the test suite JVM execution has been specifically prepared to be Scala.js standard).

Reply:
Ah, yes, I forgot to explicitly discuss that. It's true that so far we've specified Scala.js as en-US. If that is still the case, then yes this should be EnglishUS. But the reason I've said it was specified as en-US all along was because I did not know there was such a thing as the Root locale. And now that I know that it exists, I think we should redefine Scala.js as using the Root locale by specification.

In practice, this changes virtually nothing, because Root is basically the same as en, however it slightly differs from en-US, notably for currencies (the currency of Root and of en is `XXX`, while that of en-US is `USD`). So far we've never provided anything currency-oriented, though.

So why pick ROOT instead of the (conservative) US? Because it's less biased. It corresponds to all the algorithms that are specified as *locale-insensitive*. So in that sense it is more canonical than en-US, which is a biased choice.

Reply:
@cquiroz What do you think about the rationale above?

Reply:
Ok, fair enough that sounds reasonable. But can one actually configure a JVM to run in the Root locale? If yes, we should do so for our test suite, if no, it's a bit hairier :S

Reply:
It turns out that this is possible, by given explicit but empty properties for `-Duser.language` and `-Duser.country`.